context.go

```diff
@@ -0,0 +1,43 @@
+package stats
+
+import "context"
+
+func ContextWithTags(ctx context.Context, tags ...Tag) context.Context {
+	if x := getTagSlice(ctx); x != nil {
+		*x = append(*x, tags...)
+		return ctx
+	}
+	// initialize the context reference and return a new context
+	return context.WithValue(ctx, contextKeyReqTags, &tags)
+}
+
+func ContextTags(ctx context.Context) []Tag {
+	if x := getTagSlice(ctx); x != nil {
+		ret := make([]Tag, len(*x))
+		copy(ret, *x)
+		return ret
+	}
+	return nil
+}
+
+func getTagSlice(ctx context.Context) *[]Tag {
+	if tags, ok := ctx.Value(contextKeyReqTags).(*[]Tag); ok {
+		return tags
+	}
+	return nil
```

Comment:
This function can be simplified to leverage the zero-value of a pointer type:
```go
tags, _ := ctx.Value(contextKeyReqTags).(*[]Tag)
```
Which also makes its existence questionable.

Reply:
Yeah, but this is slightly more clear.  I don't think it needs to be simplified unless you think it'll result in some optimizations or something.